Gems/GraphCanvas/Code/StaticLib/GraphCanvas/Widgets/NodePalette/Model/NodePaletteSortFilterProxyModel.cpp

```diff
@@ -187,6 +187,57 @@ namespace GraphCanvas
         return showRow;
     }
 
+    bool NodePaletteSortFilterProxyModel::lessThan(const QModelIndex& source_left, const QModelIndex& source_right) const
+    {
+        QAbstractItemModel* model = sourceModel();
+        QString left = model->data(source_left).toString();
+        QString right = model->data(source_right).toString();
+
+        if (m_filter.isEmpty())
+        {
+            return left < right;
+        }
+        else
+        {
+            int leftScore = CalculateSortingScore(source_left);
+            int rightScore = CalculateSortingScore(source_right);
+            // When sorting score is equal, sort based on alphabetical order
+            if (leftScore == rightScore)
+            {
+                return left < right;
+            }
+            else
+            {
+                return leftScore < rightScore;
+            }
+        }
+    }
+
+    int NodePaletteSortFilterProxyModel::CalculateSortingScore(const QModelIndex& source) const
+    {
+        QAbstractItemModel* model = sourceModel();
+        int result = INT_MAX;
+        if (model->hasChildren(source))
+        {
+            for (int i = 0; i < model->rowCount(source); ++i)
+            {
+                QModelIndex child = model->index(i, 0, source);
+                result = AZStd::min(result, CalculateSortingScore(child));
+            }
+        }
+        QString sourceString = model->data(source).toString();
+
+        if (sourceString.compare(m_filter, Qt::CaseInsensitive) == 0)
```

Comment:
Move this check up, right after you initialize the model so that a match is an early return past the other, unused logic in this case.